pkg/cloud/gcp/actuators/util/gcp_uefi_disk_check.go

```diff
@@ -0,0 +1,131 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	computeservice "github.com/openshift/machine-api-provider-gcp/pkg/cloud/gcp/actuators/services/compute"
+)
+
+const (
+	UEFICompatible = "UEFI_COMPATIBLE"
+)
+
+// IsUEFICompatible detects if the machine's boot disk was created with a UEFI image.
+// Shielded VMs require UEFI-compatible images. In some instances customers may still
+// be using non UEFI-compatible images. e.g OpenShift images listed on the GCP marketplace
+// are not updated with every release, and the 4.8 image is used until 4.12 and was not
+// created with UEFI support.
+func IsUEFICompatible(gceService computeservice.GCPComputeService, providerConfig *machinev1.GCPMachineProviderSpec) (bool, error) {
+	for _, disk := range providerConfig.Disks {
+		if !disk.Boot {
+			continue
+		}
+
+		// Parse the image reference from the disk.
+		imageRef, err := parseImageReference(disk.Image, providerConfig.ProjectID)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse disk image reference: %w", err)
+		}
+
+		if imageRef.IsFamily {
+			return checkImageFamilyImageUEFICompatible(gceService, imageRef.Project, providerConfig.Zone, imageRef.Image)
+		}
+		return checkImageUEFICompatible(gceService, imageRef.Project, imageRef.Image)
+	}
+	return false, fmt.Errorf("no boot disk found")
+}
+
+// imageReference holds parsed details from a disk.Image string.
+type imageReference struct {
+	Project  string
+	Image    string
+	IsFamily bool // true if the image is specified as a family reference.
+}
+
+// parseImageReference extracts project and image information from the given image string.
+// It supports various formats:
+//   - "projects/{project}/global/images/{image}"
+//   - "projects/{project}/global/images/family/{imageFamily}"
+//   - A simple image name without slashes, in which case providerProject is used.
+func parseImageReference(imageStr, providerProject string) (*imageReference, error) {
+	// If the image string does not contain a slash, assume it's a simple image name.
+	if !strings.Contains(imageStr, "/") {
+		return &imageReference{
+			Project:  providerProject,
+			Image:    imageStr,
+			IsFamily: false,
+		}, nil
+	}
+
+	// Check if the image string contains "projects/".
+	if !strings.Contains(imageStr, "projects/") {
```

Comment:
Are you doing `Contains` here so that it supports full URLs? You [mentioned on Slack](https://redhat-internal.slack.com/archives/GE2HQ9QP4/p1737654701440449?thread_ts=1737479970.161129&cid=GE2HQ9QP4) that full URLs are valid. If that's still correct, maybe add that to the list of supported formats and a test case for it.

Reply:
This was more to verify the format of the inputted disk image string. I think the split should strip everything before `/projects`, keeping full URLs supported, but if we don't have any `/projects`, but do have a `/` we have an unexpected input which may break the parsing below

Reply:
I also want to add tests for this function so it's really clear how it's expected to function, so will put a full URL in the cases for that.